src/core/hle/kernel/k_priority_queue.h

```diff
@@ -0,0 +1,443 @@
+        constexpr KPerCoreQueue() : root() {
+            for (size_t i = 0; i < NumCores; i++) {
+                this->root[i].Initialize();
+            }
+        }
+
+        constexpr bool PushBack(s32 core, Member* member) {
+            /* Get the entry associated with the member. */
+            Entry& member_entry = member->GetPriorityQueueEntry(core);
+
+            /* Get the entry associated with the end of the queue. */
+            Member* tail = this->root[core].GetPrev();
+            Entry& tail_entry =
+                (tail != nullptr) ? tail->GetPriorityQueueEntry(core) : this->root[core];
+
+            /* Link the entries. */
+            member_entry.SetPrev(tail);
+            member_entry.SetNext(nullptr);
+            tail_entry.SetNext(member);
+            this->root[core].SetPrev(member);
+
+            return (tail == nullptr);
+        }
+
+        constexpr bool PushFront(s32 core, Member* member) {
+            /* Get the entry associated with the member. */
+            Entry& member_entry = member->GetPriorityQueueEntry(core);
+
+            /* Get the entry associated with the front of the queue. */
+            Member* head = this->root[core].GetNext();
+            Entry& head_entry =
+                (head != nullptr) ? head->GetPriorityQueueEntry(core) : this->root[core];
+
+            /* Link the entries. */
+            member_entry.SetPrev(nullptr);
+            member_entry.SetNext(head);
+            head_entry.SetPrev(member);
+            this->root[core].SetNext(member);
+
+            return (head == nullptr);
+        }
+
+        constexpr bool Remove(s32 core, Member* member) {
+            /* Get the entry associated with the member. */
+            Entry& member_entry = member->GetPriorityQueueEntry(core);
+
+            /* Get the entries associated with next and prev. */
+            Member* prev = member_entry.GetPrev();
+            Member* next = member_entry.GetNext();
+            Entry& prev_entry =
+                (prev != nullptr) ? prev->GetPriorityQueueEntry(core) : this->root[core];
+            Entry& next_entry =
+                (next != nullptr) ? next->GetPriorityQueueEntry(core) : this->root[core];
+
+            /* Unlink. */
+            prev_entry.SetNext(next);
+            next_entry.SetPrev(prev);
+
+            return (this->GetFront(core) == nullptr);
+        }
+
+        constexpr Member* GetFront(s32 core) const {
+            return this->root[core].GetNext();
+        }
+    };
+
+    class KPriorityQueueImpl {
+    private:
+        KPerCoreQueue queues[NumPriority];
+        Common::BitSet64<NumPriority> available_priorities[NumCores];
+
+    public:
+        constexpr KPriorityQueueImpl() : queues(), available_priorities() { /* ... */
+        }
+
+        constexpr void PushBack(s32 priority, s32 core, Member* member) {
+            ASSERT(IsValidCore(core));
+            ASSERT(IsValidPriority(priority));
+
+            if (priority <= LowestPriority) {
```

Comment:
```suggestion
            if (priority > LowestPriority) {
                return;
            }
```

Allows unindenting the code a little bit. Ditto for the following two functions